django/db/models/fields/__init__.py

```diff
@@ -2408,3 +2386,150 @@ def formfield(self, **kwargs):
         }
         defaults.update(kwargs)
         return super(UUIDField, self).formfield(**defaults)
+
+
+class DateTransform(Transform):
+    extract_type = None
+
+    def as_sql(self, compiler, connection):
+        sql, params = compiler.compile(self.lhs)
+        lhs_output_field = self.lhs.output_field
+        if isinstance(lhs_output_field, DateTimeField):
```

Comment:
The `isinstance` check feels a bit dirty but I guess it's simpler than having two transform classes. No strong opinion from me, just something to think about.